### PR TITLE
fix form data and blob compat

### PR DESCRIPTION
**Pull request by calvinmetcalf**
FormData, Blob, and FileReader need to be on the self object, otherwise thinks they aren't supported

### CODE REVIEW

**Comment by stefanpenner** [comment]
@calvinmetcalf window isn't always available.

**Comment by calvinmetcalf** [comment]
ok now it should work in web workers, will get a test next

**Comment by stefanpenner** [comment]
@calvinmetcalf LGTM, thanks for catching this one!

**Comment by calvinmetcalf** [comment]
I'd be happy to add a test for this, but I'm not familiar enough with how the ember tests are setup to add the first test to this repo.

**Comment by stefanpenner** [comment]
some places to look:

* test likely could go here: https://github.com/stefanpenner/ember-fetch/blob/master/tests/acceptance/root-test.js
* if you need to fake the underlying XHR, we use [pretender](https://github.com/pretenderjs/pretender)

if you have other specific questions i can try to answer them

**Comment by calvinmetcalf** [comment]
done, though I do need to point out that qunit is terrible for async tests (also in general)

**Comment by stefanpenner** [comment]
@calvinmetcalf you can actually just return the promise. And qunit will automatically go async, and also assert fulfillment and assert against rejection.  Should cleanup stuff and make qunit more tolerable.

**Comment by calvinmetcalf** [comment]
well that certainly is an improvement over having to call stop

File: tests/acceptance/root-test.js [hunk 1]
```diff
@@ -37,3 +38,45 @@ test('visiting /', function(assert) {
...
+test('posting a string', function(assert) {
+  server.post('/upload', function(req) {
+    assert.equal(req.requestBody, 'foo');
+    return [
+      200,
+      { 'Content-Type': 'text/json'},
+      JSON.stringify({ name: 'World' })
+    ];
+  });
+  fetch('/upload', {
```
Comment (stefanpenner) [inline tests/acceptance/root-test.js]: Missing return?

File: tests/acceptance/root-test.js [hunk 2]
```diff
@@ -37,3 +38,45 @@ test('visiting /', function(assert) {
...
+    method: 'post',
+    body: 'foo'
+  }).then(function (res) {
+    assert.equal(res.status, 200);
+    return res.json();
+  }).then(function (data) {
+    assert.equal(data.name, 'World');
+  }).catch(function (e) {
+    assert(false, e);
```
Comment (stefanpenner) [inline tests/acceptance/root-test.js]: This won't be needed once return above is added

**Comment by stefanpenner** [comment]
Some last few comments. Once addressed, we can merge and I'll release

**Comment by calvinmetcalf** [comment]
fixed

**Pull request merged**